proxy/http/HttpBodyFactory.cc

```diff
@@ -457,36 +437,29 @@ const char *
 HttpBodyFactory::determine_set_by_host(HttpTransact::State *context)
 {
   const char *set;
-  RawHashTable_Value v;
   int host_len = context->hh_info.host_len;
   char host_buffer[host_len + 1];
   strncpy(host_buffer, context->hh_info.request_host, host_len);
   host_buffer[host_len] = '\0';
-  if (table_of_sets->getValue((RawHashTable_Key)host_buffer, &v)) {
-    set = table_of_sets->getKeyFromBinding(table_of_sets->getCurrentBinding((RawHashTable_Key)host_buffer));
+  if (auto it = table_of_sets->find(host_buffer); it != table_of_sets->end()) {
+    set = it->first.c_str();
   } else {
     set = "default";
   }
   return set;
 }
 
 const char *
-HttpBodyFactory::determine_set_by_language(RawHashTable *table_of_sets, StrList *acpt_language_list, StrList *acpt_charset_list,
-                                           float *Q_best_ptr, int *La_best_ptr, int *Lc_best_ptr, int *I_best_ptr)
+HttpBodyFactory::determine_set_by_language(std::unique_ptr<BodySetTable> &table_of_sets, StrList *acpt_language_list,
```

Comment:
I think it would be better to pass `BodySetTable const&` instead of `std::unique_ptr<BodySetTable>&`. You don't need the unique pointer inside this method.

Reply:
I kept the existing logic in there because previously, it was a pointer and there are some logics performing some checks.